### PR TITLE
Fix MsgStream inconsistent seek

### PR DESCRIPTION
**Pull request by xiaofan-luan**
see also #12041
Signed-off-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

File: internal/util/rocksmq/client/rocksmq/client_impl.go [hunk 1]
```diff
@@ -119,6 +119,13 @@ func (c *client) Subscribe(options ConsumerOptions) (Consumer, error) {
 
 func (c *client) consume(consumer *consumer) {
 	defer c.wg.Done()
+	// trigger first read
+	select {
+	case consumer.MsgMutex() <- struct{}{}:
+		break
+	default:
+		break
+	}
```
Comment (congqixia) [inline internal/util/rocksmq/client/rocksmq/client_impl.go:128]: break is not needed

File: internal/util/rocksmq/client/rocksmq/client_impl.go [hunk 2]
```diff
@@ -147,10 +154,22 @@ func (c *client) consume(consumer *consumer) {
...
+						return
+					}
+				}
+				// may have more messages left in the queue, try consume once
+				if len(msgs) == n {
+					select {
+					case consumer.MsgMutex() <- struct{}{}:
+						break
+					default:
+						break
```
Comment (congqixia) [inline internal/util/rocksmq/client/rocksmq/client_impl.go:172]: break is not needed

File: internal/util/rocksmq/server/rocksmq/rocksmq_impl.go [hunk 3]
```diff
@@ -663,11 +659,7 @@ func (rmq *rocksmq) Consume(topicName string, groupName string, n int) ([]Consum
 		consumedIDs = append(consumedIDs, msg.MsgID)
 	}
 	newID := consumedIDs[len(consumedIDs)-1]
-	err = rmq.seek(topicName, groupName, newID)
-	if err != nil {
-		log.Debug("RocksMQ: Seek(" + groupName + "," + topicName + "," + strconv.FormatInt(newID, 10) + ") failed")
-		return nil, err
-	}
+	rmq.moveConsumePos(topicName, groupName, newID+1)
```
Comment (fishpenguin) [inline internal/util/rocksmq/server/rocksmq/rocksmq_impl.go:662]: need an error handle.

**Pull request merged**